src/core/common/tlvs.hpp

```diff
@@ -163,6 +163,19 @@ class Tlv
         return reinterpret_cast<const Tlv *>(reinterpret_cast<const uint8_t *>(this) + GetSize());
     }
 
+    /**
+     * This method indicates whether the TLV and its Value are bounded by (i.e., `<=`) a given end pointer.
```

Comment:
```suggestion
     * This method indicates whether the TLV and its Value are bounded by (i.e., `<`) a given end pointer.
```

I think we generally want the last byte of the TLV to be before the end pointer, right?

Reply:
Yes. Indeed. Thanks.
The implementing is correct in using `<=` (since we go to the next TLV which is after end of current one) but considering current TLV itself we should expect all its bytes are strictly before  (`<`) the given end pointer.